src/Microsoft.Health.Fhir.Liquid.Converter.Tool/TemplateManagementLogicHandler.cs

```diff
@@ -0,0 +1,90 @@
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Tool
+{
+    internal static class TemplateManagementLogicHandler
+    {
+        private static readonly ILogger Logger;
+        private static readonly TextWriter ErrorWriter;
+
+        static TemplateManagementLogicHandler()
+        {
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddFilter("Microsoft.Health.Fhir.TemplateManagement", LogLevel.Trace)
+                       .AddConsole();
+            });
+            FhirConverterLogging.LoggerFactory = loggerFactory;
+            Logger = FhirConverterLogging.CreateLogger(typeof(TemplateManagementLogicHandler));
+
+            ErrorWriter = Console.Error;
+        }
+
+        internal static async Task PullAsync(PullTemplateOptions options)
+        {
+            try
+            {
+                if (!options.ForceOverride)
+                {
+                    if (Directory.Exists(options.OutputTemplateFolder) && Directory.GetFileSystemEntries(options.OutputTemplateFolder).Length != 0)
+                    {
+                        Logger.LogError("The Output folder is not empty. If force to override, please add -f in parameters");
+                        return;
+                    }
+                }
+
+                OCIFileManager fileManager = new OCIFileManager(options.ImageReference, options.OutputTemplateFolder);
+                if (await fileManager.PullOCIImageAsync())
+                {
+                    fileManager.UnpackOCIImage();
+                    Logger.LogInformation($"Succeed to pull templates to {options.OutputTemplateFolder} folder");
+                }
+                else
+                {
+                    Logger.LogError("Process failed");
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError("Fail to pull templates.");
+                var error = new ConverterError(ex);
+                ErrorWriter.WriteLine(JsonConvert.SerializeObject(error));
+            }
+        }
+
+        internal static async Task PushAsync(PushTemplateOptions options)
+        {
+            try
+            {
+                OCIFileManager fileManager = new OCIFileManager(options.ImageReference, options.InputTemplateFolder);
+                fileManager.PackOCIImage(options.BuildNewBaseLayer);
+                if (await fileManager.PushOCIImageAsync())
+                {
+                    Logger.LogInformation($"Succeed to push new templates to {options.ImageReference}");
+                }
+                else
+                {
+                    Logger.LogError("Process failed");
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError("Fail to push templates.");
+                var error = new ConverterError(ex);
+                TextWriter errorWriter = Console.Error;
```

Comment:
Maybe we can directly use the `ErrorWriter`?

Reply:
Thanks! fix it.